dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/ModuleSelectionConfigImpl.java

```diff
@@ -0,0 +1,455 @@
+            for (Iterator<ModuleName> iterator = includedNotYetSeen.iterator(); iterator.hasNext();) {
+                ModuleName moduleName = iterator.next();
+                for (ModuleInfo moduleInfo : moduleInfos) {
+                    if (matches(moduleName, moduleInfo, false)) {
+                        includedNotYetSeenButSeenUnderOldNaming.put(moduleName,
+                                                                    moduleInfo.getApplicationInfo().getDeploymentName());
+                        iterator.remove();
+                        break;
+                    }
+                }
+            }
+
+            for (ModuleName unmatchedInclude : includedNotYetSeenButSeenUnderOldNaming.keySet()) {
+                Tr.warning(tc, MessageConstants.OPENAPI_USING_WRONG_NAME_SOURCE_CWWKO1680W, "includeApplication or includeModule", unmatchedInclude.toString(),
+                           includedNotYetSeenButSeenUnderOldNaming.get(unmatchedInclude));
+            }
+        }
+
+        for (String unmatchedInclude : includedNotYetSeen.stream().map(ModuleName::toString).collect(toList())) {
+            Tr.warning(tc, MessageConstants.OPENAPI_MERGE_UNUSED_INCLUDE_CWWKO1667W, Constants.MERGE_INCLUDE_CONFIG, unmatchedInclude);
+        }
+
+        //Now we repeat the process for excluded applications and modules, however this time we only throw a warning message if we get a match on the deployment descriptor's name
+
+        List<ModuleName> excludedNotYetSeen = new ArrayList<>(getConfigValues().excluded);
+        for (Iterator<ModuleName> iterator = excludedNotYetSeen.iterator(); iterator.hasNext();) {
+            ModuleName moduleName = iterator.next();
+            for (ModuleInfo moduleInfo : moduleInfos) {
+                if (matches(moduleName, moduleInfo)) {
+                    iterator.remove();
+                    break;
+                }
+            }
+        }
+
+        Map<ModuleName, String> excludedNotYetSeenButSeenUnderOldNaming = new HashMap<>();
+        if (getConfigValues().serverxmlmode && ProductInfo.getBetaEdition()) {
+            for (Iterator<ModuleName> iterator = excludedNotYetSeen.iterator(); iterator.hasNext();) {
+                ModuleName moduleName = iterator.next();
+                for (ModuleInfo moduleInfo : moduleInfos) {
+                    if (matches(moduleName, moduleInfo, false)) {
+                        excludedNotYetSeenButSeenUnderOldNaming.put(moduleName,
+                                                                    moduleInfo.getApplicationInfo().getDeploymentName());
+                        iterator.remove();
+                        break;
+                    }
+                }
+            }
+
+            for (ModuleName unmatchedInclude : excludedNotYetSeenButSeenUnderOldNaming.keySet()) {
+                Tr.warning(tc, MessageConstants.OPENAPI_USING_WRONG_NAME_SOURCE_CWWKO1680W, "excludeApplication or excludeModule", unmatchedInclude.toString(),
```

Comment:
We cannot substitute "excludeApplication or excludeModule" into a translated message.

Similarly for the include message further up in the method.